src/actions/mintArveaweNFT.ts

```diff
@@ -0,0 +1,266 @@
+import BN from 'bn.js';
+import { calculate } from '@metaplex/arweave-cost';
+import { LAMPORTS_PER_SOL, PublicKey } from '@solana/web3.js';
+import crypto from 'crypto';
+import { Connection } from '../Connection';
+import { MetadataJson } from '../types';
+import { Wallet } from '../wallet';
+import { ArweaveStorage, ArweaveUploadResult } from '../providers';
+import { PayForFiles } from '../programs';
+import { prepareTokenAccountAndMintTx } from './shared';
+import { Metadata, Creator, MetadataDataData, CreateMetadata } from '../programs/metadata';
+import { sendTransaction } from './transactions';
+import { updateMetadata } from './updateMetadata';
+import { createMasterEdition } from './createMasterEdition';
+
+export enum ENftProgress {
+  none,
+  minting,
+  preparing_assets,
+  signing_metadata_transaction,
+  sending_transaction_to_solana,
+  waiting_for_initial_confirmation,
+  waiting_for_final_confirmation,
+  uploading_to_arweave,
+  updating_metadata,
+  signing_token_transaction,
+}
+
+export async function convertFile2Hash(file: File): Promise<Buffer> {
+  const hashSum = crypto.createHash('sha256');
+  const data = await file.text();
+  hashSum.update(data);
+  const hex = hashSum.digest('hex');
+  return Buffer.from(hex);
+}
+
+export function getAssetCostInfo(files: Map<string, Buffer>) {
+  const sizes = Array.from(files.values()).map((f) => f.byteLength);
+  return calculate(sizes);
+}
+
+export function createFilePack(metadata: MetadataJson, WebFile = File): File {
```

Comment:
probably it makes sense to define interface or type for `WebFile` param

Reply:
Interface is define in `lib.dom.d.ts`

Reply:
It's defined in `lib/dom.d.ts` for nodejs env we can use `@web-std/file`